src/Graphula/Dependencies.hs

```diff
@@ -24,18 +24,29 @@ module Graphula.Dependencies
 
     -- * Non-serial keys
   , KeySourceType (..)
+  , KeySourceTypeM
+  , KeyForInsert
+  , KeyRequirementForInsert
+  , InsertWithPossiblyRequiredKey (..)
+  , Required (..)
+  , Optional (..)
```

Comment:
Do users bump into this stuff at all, or is it all encapsulated?

Reply:
The only concern is it might show up in type errors, not sure. Otherwise no.